### PR TITLE
added RunType components in EpochStart

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- added RunType components in EpochStart
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: epochStart/bootstrap/interface.go [hunk 1]
```diff
@@ -72,3 +75,9 @@ type EpochStartBootstrapper interface {
 	Close() error
 	IsInterfaceNil() bool
 }
+
+type runTypeComponentsHolder interface {
```
Comment (mariusmihaic) [inline epochStart/bootstrap/interface.go:79]: :+1:

Comment (mariusmihaic) [reply epochStart/bootstrap/interface.go:79]: We should make it exported, though

File: epochStart/bootstrap/process_test.go [hunk 2]
```diff
@@ -647,23 +647,35 @@ func TestNewEpochStartBootstrap_NilArgsChecks(t *testing.T) {
...
 		require.True(t, errors.Is(err, errorsMx.ErrNilShardCoordinatorFactory))
 	})
-	t.Run("nil additional storage service creator", func(t *testing.T) {
+	t.Run("nil AdditionalStorageServiceCreator should error", func(t *testing.T) {
 		t.Parallel()
 
 		coreComp, cryptoComp := createComponentsForEpochStart()
 		args := createMockEpochStartBootstrapArgs(coreComp, cryptoComp)
-		args.AdditionalStorageServiceCreator = nil
+		args.RunTypeComponents = &mock.RunTypeComponentsStub{ShardCoordinatorFactory: &testscommon.MultiShardCoordinatorFactoryMock{}, AdditionalStorageServiceFactory: nil}
```
Comment (mariusmihaic) [inline epochStart/bootstrap/process_test.go:677]: Same comment as in previous prs

File: epochStart/mock/runTypeComponentsStub.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+// RunTypeComponentsStub -
+type RunTypeComponentsStub struct {
+	AdditionalStorageServiceFactory process.AdditionalStorageServiceCreator
+	ShardCoordinatorFactory         sharding.ShardCoordinatorFactory
+}
+
+// NewRunTypeComponentsStub -
+func NewRunTypeComponentsStub() *RunTypeComponentsStub {
```
Comment (mariusmihaic) [inline epochStart/mock/runTypeComponentsStub.go:16]: :+1:

File: testscommon/shardCoordinatorFactoryMock.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package testscommon
+
+import (
+	"github.com/multiversx/mx-chain-go/sharding"
+)
+
+// MultiShardCoordinatorFactoryMock -
+type MultiShardCoordinatorFactoryMock struct {
```
Comment (mariusmihaic) [inline testscommon/shardCoordinatorFactoryMock.go:8]: :+1:

File: epochStart/bootstrap/interface.go (hunk 1 above)
Comment (axenteoctavian) [reply epochStart/bootstrap/interface.go:79]: why exported?

**Review by raduchis — APPROVED** [review]

**Review by mariusmihaic — DISMISSED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**